modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/SchedulePublicationMVCActionCommand.java

```diff
@@ -0,0 +1,124 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.change.tracking.web.internal.portlet.action;
+
+import com.liferay.change.tracking.constants.CTPortletKeys;
+import com.liferay.change.tracking.web.internal.scheduler.PublishScheduler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Samuel Trong Tran
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + CTPortletKeys.CHANGE_LISTS,
+		"mvc.command.name=/change_lists/reschedule_publication",
+		"mvc.command.name=/change_lists/schedule_publication",
+		"mvc.command.name=/change_lists/unschedule_publication"
+	},
+	service = MVCActionCommand.class
+)
+public class SchedulePublicationMVCActionCommand extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		Date publishDate = null;
+
+		String actionName = ParamUtil.getString(
+			actionRequest, ActionRequest.ACTION_NAME);
+
+		if (!actionName.equals("/change_lists/unschedule_publication")) {
+			int day = ParamUtil.getInteger(actionRequest, "publishTimeDay");
+			int minute = ParamUtil.getInteger(
+				actionRequest, "publishTimeMinute");
+			int month = ParamUtil.getInteger(actionRequest, "publishTimeMonth");
+			int year = ParamUtil.getInteger(actionRequest, "publishTimeYear");
+
+			int hour = ParamUtil.getInteger(actionRequest, "publishTimeHour");
+
+			if (ParamUtil.getInteger(actionRequest, "publishTimeAmPm") ==
+					Calendar.PM) {
+
+				hour += 12;
+			}
+
+			publishDate = _portal.getDate(
+				month, day, year, hour, minute, themeDisplay.getTimeZone(),
+				PortalException.class);
+
+			Date currentDate = new Date(System.currentTimeMillis());
+
+			if (!publishDate.after(currentDate)) {
+				SessionErrors.add(actionRequest, "publishTime");
+
+				_portal.copyRequestParameters(actionRequest, actionResponse);
+
+				actionResponse.setRenderParameter(
+					"mvcPath", "/change_lists/schedule_publication.jsp");
```

Comment:
I broke this part on fb79c87e7433fd3d60145aa0c2c7e6259cc6cc70. Now throws a NPE. If you need to duplicate something from SchedulePublicationMVCRenderCommand that's not a problem but I'd prefer we go back through that code instead.